src/foam/nanos/fs/FileDataDAO.js

```diff
@@ -33,6 +33,37 @@ foam.CLASS({
       name: 'put_',
       javaCode: `
       File file = (File) obj;
+      DAO fileTypeDAO = (DAO) x.get("fileTypeDAO");
+      FileType fileType;
+      String type = "";
+
+      if ( SafetyUtil.isEmpty(file.getMimeType()) ) {
```

Comment:
FileDropZone will set MimeType, but fileDAO detailed view doesn't do it, so need to check